lesson5/public/index.html

```diff
@@ -0,0 +1,90 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+    <title>Shop</title>
+</head>
+
+<style>
+    .cart {
+        width: 300px;
+        height: 500px;
+        border: 1px solid #000;
+        float: right;
+    }
+</style>
+
+<body>
+<div id="app">
+    <header>
+        <input type="text" v-model="searchQuery" class="search-text" />
+        <button class="search-button" @click="handleSearchClick">Поиск</button>
+        <button class="cart-button" @click="handleCartClick">Корзина</button>
+        <div class="cart" v-show="isVisibleCart">
+            <div v-for="item in cart" class="itemCart">
+                <h2>{{item.name}}</h2>
+                <span>{{item.quantity}} шт </span>
+                <span>{{item.price}} руб</span>
+            </div>
+            <h3>Итого: {{total()}} руб</h3>
+        </div>
+    </header>
+    <main>
+        <div class="goods" v-if="filteredItems.length != 0">
+            <div v-for="item in filteredItems" class="item">
+                <h2>{{item.name}}</h2>
+                <span>{{item.price}} руб</span>
+            </div>
+        </div>
+        <div v-else>Нет данных</div>
+    </main>
+</div>
+<script src="https://cdn.jsdelivr.net/npm/vue@2.6.10/dist/vue.min.js"></script>
+<script>
+    const app = new Vue({
+        el: '#app',
+        data: {
+            items: [],
+            filteredItems: [],
+            searchQuery: '',
+            cart: [],
+            quantity: 0,
+            isVisibleCart: true
+        },
+        mounted() {
+            fetch('http://localhost:3000/products')
+                .then(response => response.json())
+                .then((items) => {
+                    this.items = items;
+                    this.filteredItems = items;
+                }),
+                fetch('http://localhost:3000/cart')
+                    .then(response => response.json())
+                    .then((cart) => {
+                        this.cart = cart;
+                    })
+        },
+        methods: {
+            filterItems(query) {
+                const regexp = new RegExp(query, 'i');
+                this.filteredItems = this.items.filter((item) => regexp.test(item.name));
+            },
+            handleSearchClick() {
+                this.filterItems(this.searchQuery);
+            },
+            handleCartClick() {
+                if (this.isVisibleCart) {
```

Comment:
весь этот код можно заменить одной строчкой
```suggestion
                this.isVisibleCart = !this.isVisibleCart;
```